### ISSUE TITLE
Order_by not working for search

### ISSUE THREAD

**Issue by Prid13** [jikan-me/jikan#399]
Am I doing something wrong, or is the `order_by` parameter not working for the `/anime` search endpoint (with `q` applied)?

**Example**
Try going to this URL: https://api.jikan.moe/v4/anime?q=naruto&order_by=popularity

<details>
<summary>Output <sub>(q=naruto&order_by=popularity)</sub></summary>

_results stripped from unnecessary data_

```json
{
  "data": [
    {
      "Title": "Naruto",
      "Popularity": 8
    },
    {
      "Title": "Boruto: Naruto the Movie - Naruto ga Hokage ni Natta Hi",
      "Popularity": 1750
    },
    {
      "Title": "Naruto: Honoo no Chuunin Shiken! Naruto vs. Konohamaru!!",
      "Popularity": 2161
    },
    {
      "Title": "Naruto Soyokazeden Movie: Naruto to Mashin to Mitsu no Onegai Dattebayo!!",
      "Popularity": 2662
    },
    {
      "Title": "Naruto: Shippuuden",
      "Popularity": 16 // <---- wrong
    },
   ...
  ]
}
```
</details>

The `order_by=popularity` parameter is completely ignored. Compare it to the normal request: https://api.jikan.moe/v4/anime?q=naruto

<details>
<summary>Output <sub>(q=naruto)<sub></summary>

```json
{
  "data": [
    {
      "Title": "Naruto",
      "Popularity": 8
    },
    {
      "Title": "Boruto: Naruto the Movie - Naruto ga Hokage ni Natta Hi",
      "Popularity": 1750
    },
    {
      "Title": "Naruto: Honoo no Chuunin Shiken! Naruto vs. Konohamaru!!",
      "Popularity": 2161
    },
    {
      "Title": "Naruto Soyokazeden Movie: Naruto to Mashin to Mitsu no Onegai Dattebayo!!",
      "Popularity": 2662
    },
    {
      "Title": "Naruto: Shippuuden",
      "Popularity": 16
    }
  ...
  ]
}
```
</details>

They're exactly the same!

I can't get `sort` to work either. And `sfw` is a bit peculiar to me as it's set to filter out adult content, yet series like Highschool DxD and Sekirei are perfectly searchable from the front page of your website even. Or is it only for anime with _covers_ that are super explicit?

Anyways, I'm asking this because the default search results returned are a bit "random" and don't necessarily reflect the best results first. Which is why I'm trying to order them by popularity to see if I can get "Naruto" and "Naruto Shippuden" among the top results when searching for "naruto" 😅

Thanks :)

**Comment by davdi1337** [comment]
Try to replace the `q` endpoint with `letter` endpoint, thats seems work to me.

Example: https://api.jikan.moe/v4/anime?letter=naruto&order_by=popularity

The `sort` also works too with this endpoint.

**Comment by Prid13** [comment]
Thanks, it works! :) The only downside is that `letter` only looks for anime _starting_ with the search query. Searching "academia" for "Boku no Hero Academia" works with `q`, but doesn't with `letter`.

Appreciate the workaround, though 😇⭐

**Comment by Ervie** [comment]
Yup, this a workaround for now. It seems that right now order works only when query is empty (so it's either working `q` or working `order_by`).
